backend/aethel_db/views.py

```diff
@@ -112,6 +114,74 @@ def item_contains_query_string(item: LexicalItem) -> bool:
         return response_object.json_response()
 
 
+class AethelDetailError(Enum):
+    NO_QUERY_INPUT = "NO_QUERY_INPUT"
+    SAMPLE_NOT_FOUND = "SAMPLE_NOT_FOUND"
+    MULTIPLE_FOUND = "MULTIPLE_FOUND"
+
+
+aethelDetailStatusCodes = {
+    AethelDetailError.NO_QUERY_INPUT: status.HTTP_400_BAD_REQUEST,
+    AethelDetailError.SAMPLE_NOT_FOUND: status.HTTP_404_NOT_FOUND,
+    AethelDetailError.MULTIPLE_FOUND: status.HTTP_500_INTERNAL_SERVER_ERROR
+}
+
+@dataclass
+class AethelDetailResult:
+    sentence: str
+    name: str
+    term: str
+    subset: str
+    phrases: list[dict]
+
+@dataclass
+class AethelDetailResponse:
+    result: Optional[AethelDetailResult] = None
+    error: Optional[AethelDetailError] = None
+
+    def parse_sample(self, sample: Sample) -> None:
+        self.result = AethelDetailResult(
+            sentence=sample.sentence,
+            name=sample.name,
+            term=str(sample.proof.term),
+            subset=sample.subset,
+            phrases=serialize_phrases_with_infix_notation(sample.lexical_phrases),
+        )
+
+    def json_response(self) -> JsonResponse:
+        result = asdict(self.result) if self.result else None
+        status_code = aethelDetailStatusCodes[self.error] if self.error else status.HTTP_200_OK
+
+        return JsonResponse(
+            {
+                "result": result,
+                "error": self.error,
+            },
+            status=status_code,
+        )
+
+
 class AethelDetailView(APIView):
-    def get(self, request: HttpRequest, pk: int):
-        return Response("Success!", status=status.HTTP_200_OK)
+    def get(self, request: HttpRequest) -> JsonResponse:
+        query_input = self.request.query_params.get("sample-name", None)
+
+        if query_input is None:
+            response = AethelDetailResponse(error=AethelDetailError.NO_QUERY_INPUT)
+            return response.json_response()
+
+        samples = dataset.find_by_name(query_input)
+
+        if len(samples) == 0:
+            response = AethelDetailResponse(error=AethelDetailError.SAMPLE_NOT_FOUND)
+            return response.json_response()
+
+        if len(samples) > 1:
```

Comment:
I'd say it's the responsibility of the data layer (and not the query interface) to make sure names are unique

Reply:
I agree, but can we rely on this?

Would you rather just use `samples = dataset.find_by_name(query_input)[0]` and assume we will ever only get one result?